tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,966 @@
+
+    # Copy from project directory to output directory if not done already.
+    if not os.path.exists(args.output_directory):
+        shutil.copytree(args.project_directory, args.output_directory)
+
+    # Extract all of the kernel parameter and template type information.
+    if args.add_static_casts:
+        KernelTemplateParams = {}
+        for (dirpath, _dirnames, filenames) in os.walk(args.output_directory):
+            for filename in filenames:
+                if filename_ends_with_extension(filename, args.extensions) and inside_included_directories(dirpath, args.output_directory, args.include_dirs):
+                    the_file = os.sep.join([dirpath, filename])
+
+                    # Store param information inside KernelTemplateParams
+                    get_kernel_template_params(the_file, KernelTemplateParams)
+
+    # Open YAML file with disable information.
+    if args.yaml_settings != "":
+        with openf(args.yaml_settings, "r") as f:
+            yaml_data = yaml.load(f)
+
+        # Disable functions in certain files according to YAML description
+        for disable_info in yaml_data["disabled_functions"]:
+            filepath = os.path.join(args.output_directory, disable_info["path"])
+            if "functions" in disable_info:
+                functions = disable_info["functions"]
+            else:
+                functions = []
+
+            if "non_hip_functions" in disable_info:
+                non_hip_functions = disable_info["non_hip_functions"]
+            else:
+                non_hip_functions = []
+
+            if "non_device_functions" in disable_info:
+                not_on_device_functions = disable_info["non_device_functions"]
+            else:
+                not_on_device_functions = []
+
+            with openf(filepath, "r+") as f:
+                txt = f.read()
+                for func in functions:
+                    # TODO - Find fix assertions in HIP for device code.
+                    txt = disable_function(txt, func, disablefuncmode.ASSERT)
+
+                for func in non_hip_functions:
+                    # Disable this function on HIP stack
+                    txt = disable_function(txt, func, disablefuncmode.HCC_MACRO)
+
+                for func in not_on_device_functions:
+                    # Disable this function when compiling on Device
+                    txt = disable_function(txt, func, disablefuncmode.DEVICE_MACRO)
+
+                f.seek(0)
+                f.write(txt)
+                f.truncate()
+                f.close()
+
+        # Disable modules
+        disable_modules = yaml_data["disabled_modules"]
+        for module in disable_modules:
+            disable_module(os.path.join(args.output_directory, module))
+
+        # Disable unsupported HIP functions
+        for disable in yaml_data["disable_unsupported_hip_calls"]:
+            filepath = os.path.join(args.output_directory, disable["path"])
+            if "functions" in disable:
```

Comment:
A more idiomatic way to spell this is

```
functions = disable.get("functions", [])
```

ditto below

Reply:
done